contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,661 @@
+            .scriptBytecodeAddresses[dependencyType.scriptCount - 1]
+            .purgeBytecode();
+        // delete reference to contract address that no longer exists
+        delete dependencyType.scriptBytecodeAddresses[
+            dependencyType.scriptCount - 1
+        ];
+        unchecked {
+            dependencyType.scriptCount = dependencyType.scriptCount - 1;
+        }
+
+        emit DependencyTypeScriptUpdated(_dependencyTypeId);
+    }
+    
+    function updateDependencyTypePreferredCDN(
+        bytes32 _dependencyTypeId,
+        string memory _preferredCDN
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredCDN.selector)
+        onlyNonEmptyString(_preferredCDN)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredCDN = _preferredCDN;
+
+        emit DependencyTypePreferredCDNUpdated(_dependencyTypeId, _preferredCDN);
+    }
+
+    function updateDependencyTypePreferredRepository(
+        bytes32 _dependencyTypeId,
+        string memory _preferredRepository
+    )
+        external
+        onlyAdminACL(this.updateDependencyTypePreferredRepository.selector)
+        onlyNonEmptyString(_preferredRepository)
+    {
+        dependencyTypeInfo[_dependencyTypeId].preferredRepository = _preferredRepository;
+
+        emit DependencyTypePreferredRepositoryUpdated(
+            _dependencyTypeId,
+            _preferredRepository
+        );
+    }
+
+    function updateDependencyTypeProjectWebsite(
+        bytes32 _dependencyTypeId,
+        string memory _projectWebsite
+    ) external onlyAdminACL(this.updateDependencyTypeProjectWebsite.selector) {
+        dependencyTypeInfo[_dependencyTypeId].projectWebsite = _projectWebsite;
+
+        emit DependencyTypeProjectWebsiteUpdated(
+            _dependencyTypeId,
+            _projectWebsite
+        );
+    }
+
+    function addDependencyTypeAdditionalCDN(
+        bytes32 _dependencyTypeId,
+        string memory _additionalCDN
+    )
+        external
+        onlyAdminACL(this.addDependencyTypeAdditionalCDN.selector)
+        onlyNonEmptyString(_additionalCDN)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalCDNCount;
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNs[
+            additionalCDNCount
+        ] = _additionalCDN;
+        dependencyTypeInfo[_dependencyTypeId].additionalCDNCount =
+            additionalCDNCount +
+            1;
+
+        emit DependencyTypeAdditionalCDNUpdated(
+            _dependencyTypeId,
+            _additionalCDN,
+            additionalCDNCount
+        );
+    }
+
+    function removeDependencyTypeAdditionalCDNAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index
+    )
+        external
+        onlyAdminACL(this.removeDependencyTypeAdditionalCDNAtIndex.selector)
+        onlyExistingDependencyType(_dependencyTypeId)
+    {
+        uint24 additionalCDNCount = dependencyTypeInfo[_dependencyTypeId]
```

Comment:
Following a similar pattern to the flex contract here so that we never have a sparse "array". I don't think order should ever matter here.

Reply:
++ to following that pattern.
nit - I think would be better & more efficient to do all math with uint256, then cast as uint24 in operation
```
--- at top of contract
using SafeCast for uint24;
--- below
dependencyTypeInfo[_dependencyTypeId]
            .additionalCDNCount = uint24(lastElementIndex);
``` 
(also, since this only decrements uint24, would be fine to not even use the SafeCast library)